pkg/install/v1alpha1/cstor_pool.go

```diff
@@ -200,11 +200,13 @@ spec:
       selector:
         matchLabels:
           app: cstor-pool
+          openebs.io/version: {{ .CAST.version }}
```

Comment:
why do we need here when version is already part of `metadata.labels`?

Reply:
@vishnuitta -- labels present in `matchLabels` are those which should be matched with labels present in metadata for a controller e.g. deployment to manage the pods.
Now we need to see do we also need the version labels as part of this matching.

Reply:
Reason to add openebs.io/version labels under `matchLabels` is 1) Deployment controller will first delete the (rs, pods) which doesn't have the matching label in `.spec.template.metadata.labels`(useful in update scenarios). 
2) Deployment controller will never allow to spawn the pods if matching label value is not found in `template.metadata.labels`.